.github/workflows/test-branch.yml

```
@@ -0,0 +1,43 @@
+name: Test branch
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        ocaml-compiler:
+          # Don't include every versions. OCaml-CI already covers that
+          - 4.13.x
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      # Clone the project
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # Setup
+      - name: Setup OCaml ${{ matrix.ocaml-version }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - name: Opam dependencies
+        run: opam install --deps-only -t .
+
+      - name: Test conventional profile
+        run: ./tools/test_branch.sh -a ${{ github.event.pull_request.base.ref }} -b ${{ github.event.pull_request.head.ref }} 'profile=conventional'
```

```suggestion
        run: ./tools/test_branch.sh -a ${{ github.event.pull_request.base.ref }} -b HEAD 'profile=conventional'
```
Branch names might not be meaning full.